pkg/client/init.go

```diff
@@ -3,22 +3,26 @@ package client
 import (
 	"errors"
 	"fmt"
+	"os"
+	"strings"
+
 	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
 	kubeclient "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/klog/v2"
-	"os"
 )
 
 var (
-	kubernetesRestConfig               *rest.Config
-	kubernetesApiConfig                *clientcmdapi.Config
-	inClusterClient                    kubeclient.Interface
-	karmadaRestConfig                  *rest.Config
-	karmadaApiConfig                   *clientcmdapi.Config
-	inClusterKarmadaClient             karmadaclientset.Interface
-	inClusterClientForKarmadaApiServer kubeclient.Interface
+	kubernetesRestConfig                  *rest.Config
+	kubernetesApiConfig                   *clientcmdapi.Config
+	inClusterClient                       kubeclient.Interface
+	karmadaRestConfig                     *rest.Config
+	karmadaApiConfig                      *clientcmdapi.Config
+	inClusterKarmadaClient                karmadaclientset.Interface
+	inClusterClientForKarmadaApiServer    kubeclient.Interface
+	inClusterClientForKarmadactlApiServer kubeclient.Interface
+	clusterHost                           string
```

Comment:
the reason for we need to add `clusterHost` is that in the implementation, we modify the `rest.Config` for karmada. I think we should not modify the `rest.Config` for karmada, the recommend way is deepcopy of `rest.Config`.